### PR TITLE
docs(samples): add list configs

### PR DESCRIPTION
**Pull request by pmakani**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [X] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigquerydatatransfer/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [X] Ensure the tests and linter pass
- [X] Code coverage does not decrease (if any source code was changed)
- [X] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: samples/snippets/src/main/java/com/example/bigquerydatatransfer/ListTransferConfigs.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      ProjectName parent = ProjectName.of(projectId);
+      ListTransferConfigsRequest request =
+          ListTransferConfigsRequest.newBuilder().setParent(parent.toString()).build();
+      dataTransferServiceClient
+          .listTransferConfigs(request)
+          .iterateAll()
+          .forEach(config -> System.out.printf("Success! Config ID : %s", config.getName()));
+    } catch (IOException | ApiException ex) {
+      System.out.print("Config list not found due to error." + ex.toString());
+    }
```
Comment (lesv) [inline samples/snippets/src/main/java/com/example/bigquerydatatransfer/ListTransferConfigs.java:46]: Ok, but typically, we don't catch exceptions and let them propagate - if we don't handle them.
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md#exception-handling

File: samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_CLOUD_PROJECT");
+  }
+
+  @Before
+  public void setUp() {
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    System.setOut(out);
```
Comment (lesv) [inline samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java:66]: You might wish to capture the existing **Out** here

File: samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            .setParams(Struct.newBuilder().putAllFields(params).build())
+            .setSchedule("every 24 hours")
+            .build();
+    CreateScheduledQuery.createScheduledQuery(PROJECT_ID, transferConfig);
+    String result = bout.toString();
+    name = result.substring(result.indexOf(".") + 1);
+
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    System.setOut(out);
```
Comment (lesv) [inline samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java:101]: And restore **Out** here

File: samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java (hunk 2 above)
Comment (lesv) [reply samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java:66]: better would be to do it in a` @Before `

File: samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java (hunk 3 above)
Comment (lesv) [reply samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java:101]: And restore in an `@After`

File: samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java (hunk 2 above)
Comment (stephaniewang526) [reply samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java:66]: This is in `@Before`

Comment (lesv) [reply samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java:66]: No before is the right place - But it should still capture Out and restore it

Comment (stephaniewang526) [reply samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java:66]: We're staying consistent with other samples. e.g. https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/automl/cloud-client/src/test/java/com/example/automl/ImportDatasetTest.java

Please provide us with an example if this is not satisfactory.

Comment (lesv) [reply samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java:66]: And that's the problem - other samples are doing it wrong - we just had a case where someone was doing this and then a new test was added and they no longer had access to stdout.  Restoring the old value works.

Comment (lesv) [reply samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java:66]: The maven plugin running the test is providing stdout

Comment (stephaniewang526) [reply samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java:68]: Done

File: samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            .setDisplayName(displayName)
+            .setDataSourceId("scheduled_query")
+            .setParams(Struct.newBuilder().putAllFields(params).build())
+            .setSchedule("every 24 hours")
+            .build();
+    CreateScheduledQuery.createScheduledQuery(PROJECT_ID, transferConfig);
+    String result = bout.toString();
+    name = result.substring(result.indexOf(".") + 1);
+
+    bout = new ByteArrayOutputStream();
```
Comment (lesv) [inline samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java:101]: I don't think this is required anymore -- sorry for not seeing it earlier

File: samples/snippets/src/main/java/com/example/bigquerydatatransfer/ListTransferConfigs.java (hunk 1 above)
Comment (stephaniewang526) [reply samples/snippets/src/main/java/com/example/bigquerydatatransfer/ListTransferConfigs.java:46]: I think here we can bubble up IOException but we should catch ApiException and print out to user.

File: samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java (hunk 4 above)
Comment (stephaniewang526) [reply samples/snippets/src/test/java/com/example/bigquerydatatransfer/ListTransferConfigsIT.java]: Done. Thank you!

**Comment by stephaniewang526** [comment]
Fixed some issues in #306

**Pull request closed without merging**